Core/HLE/proAdhoc.cpp

```diff
@@ -949,41 +951,45 @@ void AfterMatchingMipsCall::SetContextID(u32 ContextID, u32 eventId) {
 void notifyAdhocctlHandlers(u32 flag, u32 error) {
 	__UpdateAdhocctlHandlers(flag, error);
 	// TODO: We should use after action instead of guessing the time like this
-	sleep_ms(20); // Ugly workaround to give time for the mips callback to fully executed, usually only need <16ms
+	//sleep_ms(20); // Ugly workaround to give time for the mips callback to fully executed, usually only need <16ms
```

Comment:
@ANR2ME  Why was this commented out?

Reply:
because it was delayed to make sure previous callback is already fully executed, but how much time a callback required to be done is just a guess, so i decide to use After action instead t be exact, but later i decide to give the delay (hleDelayResult) within `__NetTriggerCallbacks` where it process a single callback once per HLE call instead of multiple callbacks.